flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+
+<!-- skip -->
+{% prettify dart %}
+Map coordinates = await Navigator.of(context).pushNamed('/location');
+{% endprettify %}
+
+then inside your location route once the user has selected their location you 
+can "pop" the stack with the result
+
+<!-- skip -->
+{% prettify dart %}
+Navigator.of(context).pop({"lat":43.821757,"long":-79.226392});
+{% endprettify %}
+
+# Async UI
+
+**What is the equivalent of runOnUiThread in Flutter?**
+
+Because of Flutters reactive architecture, there is no need to worry about if 
+you are running stuff on the UI thread, as long as you make sure you don't 
+block your network calls.
+
+For example, when making long running calls you should always follow the 
+async/await pattern
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  String dataURL = "https://jsonplaceholder.typicode.com/posts";
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = JSON.decode(response.body);
+  });
+}
+{% endprettify %}
+
+To update the UI you would call setState which would trigger the build method 
+to run again and update the data.
+
+Here is the full example of loading data asynchronously and displaying it in a 
+ListView.
+
+<!-- skip -->
+{% prettify dart %}
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+void main() {
+  runApp(new SampleApp());
+}
+
+class SampleApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Sample App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new SampleAppPage(),
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  List widgets = new List();
+
+  @override
+  void initState() {
+    super.initState();
+
+    loadData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(
+          title: new Text("Sample App"),
+        ),
+        body: new ListView.builder(
+            itemCount: widgets.length,
+            itemBuilder: (BuildContext context, int position) {
+              return getRow(position);
+            }));
+  }
+
+  Widget getRow(int i) {
+    return new Padding(padding: new EdgeInsets.all(10.0), child: new Text("Row ${widgets[i]["title"]}"));
+  }
+
+  loadData() async {
+    String dataURL = "https://jsonplaceholder.typicode.com/posts";
+    http.Response response = await http.get(dataURL);
+    setState(() {
+      widgets = JSON.decode(response.body);
+    });
+  }
+}
+{% endprettify %}
+
+
+
+**What is the equivalent of an AsyncTask or IntentService on Android?**
+
+Often, you do not have to worry about threading in Flutter as long as 
```

Comment:
you may want to draw a distinction between "worrying about threading":

* you don't have to worry about mutable data and multiple threads, because Dart is single threaded. Yay, easier to understand!

* you DO need to worry about what you are running on the UI thread, though, even in Flutter. If you need to run work not on the UI thread, you need to run that work in an isolate.

Reply:
made it mroe clear